firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigFetchHandler.java

```diff
@@ -189,21 +189,32 @@ public ConfigFetchHandler(
                   createThrottledMessage(backoffEndTime.getTime() - currentTime.getTime()),
                   backoffEndTime.getTime()));
     } else {
-      Task<InstanceIdResult> instanceIdTask = firebaseInstanceId.getInstanceId();
+      Task<String> installationIdTask = firebaseInstallations.getId();
+      Task<InstallationTokenResult> installationTokenTask = firebaseInstallations.getToken(false);
       fetchResponseTask =
-          instanceIdTask.continueWithTask(
-              executor,
-              (completedIidTask) -> {
-                if (!completedIidTask.isSuccessful()) {
-                  return Tasks.forException(
-                      new FirebaseRemoteConfigClientException(
-                          "Failed to get Firebase Instance ID token for fetch.",
-                          completedIidTask.getException()));
-                }
-
-                InstanceIdResult instanceIdResult = completedIidTask.getResult();
-                return fetchFromBackendAndCacheResponse(instanceIdResult, currentTime);
-              });
+          Tasks.whenAllComplete(installationIdTask, installationTokenTask)
+              .continueWithTask(
+                  executor,
+                  (completedInstallationsTasks) -> {
+                    if (!installationIdTask.isSuccessful()) {
+                      return Tasks.forException(
+                          new FirebaseRemoteConfigClientException(
+                              "Failed to get Firebase Installation ID for fetch.",
+                              installationIdTask.getException()));
+                    }
+
+                    if (!installationTokenTask.isSuccessful()) {
+                      return Tasks.forException(
+                          new FirebaseRemoteConfigClientException(
+                              "Failed to get Firebase Installation token for fetch.",
+                              installationTokenTask.getException()));
+                    }
+
+                    String installationId = installationIdTask.getResult();
+                    String installationToken = installationTokenTask.getResult().getToken();
+                    return fetchFromBackendAndCacheResponse(
+                        installationId, installationToken, currentTime);
```

Comment:
I know we already talked about this, but I strongly recommend to only upload the FIS auth token, because the auth token contains all the information needed on server side, particularly the FID.
Thus, uploading both values is redundancy and might lead to inconsistencies on server side.

As far as I understood our conversations, you are planning to get rid of uploading the FID in the future, but want to upload it for now to keep the current processes consistent.
That is absolutely fine. I just wanted to mention it again.

Reply:
Thanks for calling it out!